GaloisInc/BESSPIN-Tool-Suite
# Give researchers uart access

Implements #595 
Still needs to be thoroughly tested. Testing it is not trivial as this has to run in production mode, and should be accessed from another instance. 

I will write here the testing procedure:
- Make sure you follow the instructions in `docs/remoteCommunicationWithTarget.md` to setup both of your instances.
- If you used your instance before, please use `./build/clearNetworkSetup.sh <secondIP>` first.
- In your config file, make sure you have `mode=production`.
- In `fett/base/utils/misc.py`, add the following after the queues and S3 names:
```
_settings['awsProdSqsQueueTX'] = 'https://sqs.us-west-2.amazonaws.com/363527286999/test-ticket215-TX.fifo'
_settings['awsProdSqsQueueRX'] = 'https://sqs.us-west-2.amazonaws.com/363527286999/test-ticket215-RX' 
_settings['awsProdS3Bucket'] = 'test-fett-target'
```
- Download: `aws s3 cp s3://test-fett-target/testProduction.py .`. This a utility to use to handle the queues and such and all portal communication (fake portal communication).
- To run the tool, you have to use (the job ID has to be different each time, that's why I use the timestamp for testing):
```
./fett.py -c <yourConfig.ini> -d -ep awsProd -job `date +%s`
```
- After the tool sends the deployment msg, use `./testProduction.py -s recDeployment` to delete the msg.
- To terminate, use `./testProduction -s sndTerm`. Then, after termination, use `./testProduction -s recTerm` to delete it.
- For UART access, on another instance, use `nc <secondIP> 10234` and you should be in.
